src/devices/SoftwareSpi/SoftwareSpi.cs

```diff
@@ -202,26 +268,20 @@ private class ScopeData
             internal Action _enter;
             internal Action _exit;
 
-            public ScopeData(Action enter = null, Action exit = null)
+            public ScopeData(Action enter, Action exit)
             {
-                _enter = enter;
-                _exit = exit;
+                _enter = enter ?? throw new ArgumentNullException(nameof(enter));
```

Comment:
consider `?? () => {}` instead - since lack of enter/exit action is equivalent to NOP

Reply:
I'm still curious why we didn't get NullRef before since it was lacking one of the actions in one place

Reply:
Sure I can change that, I was just thinking it would indicate an incorrect use, but it's fine, I've changed it.